packages/app/src/OAuthPage.tsx

```diff
@@ -1,20 +1,58 @@
 import { Title } from '@mantine/core';
-import { CodeChallengeMethod } from '@medplum/core';
-import { Logo, SignInForm } from '@medplum/react';
+import { CodeChallengeMethod, normalizeErrorString } from '@medplum/core';
+import { Logo, SignInForm, useMedplum } from '@medplum/react';
 import { useNavigate, useSearchParams } from 'react-router-dom';
 import { getConfig } from './config';
+import { useEffect, useState } from 'react';
+import { showNotification } from '@mantine/notifications';
+
+interface LogoInfo {
+  contentType: string;
+  url: string;
+  title: string;
+}
+
+interface ClientInfo {
+  logo: LogoInfo | null;
+  welcomeString: string | null;
+}
 
 export function OAuthPage(): JSX.Element | null {
   const navigate = useNavigate();
   const [params] = useSearchParams();
-
+  const medplum = useMedplum();
+  const scope = params.get('scope') || 'openid';
+  const [welcomeString, setWelcomeString] = useState<string | null>('Medplum');
```

Comment:
We can avoid 2 calls to `setState` if we flatten these into a single client info state:

```ts
const [clientInfo, setClientInfo] = useState<ClientApplicationSignInForm>();
```